lib/Sema/CodeSynthesis.cpp

```diff
@@ -814,6 +814,16 @@ static FuncDecl *addMaterializeForSet(AbstractStorageDecl *storage,
                              storage->getSetter());
   storage->setMaterializeForSetFunc(materializeForSet);
 
+  // Make sure we record the override.
+  //
+  // FIXME: Instead, we should just not call checkOverrides() on
+  // storage until all accessors are in place.
+  if (auto *baseASD = dyn_cast_or_null<AbstractStorageDecl>(
```

Comment:
I don't think you need the cast. We already have an `AbstractStorageDecl *getOverriddenDecl() const` method.